src/event-helper.js

```diff
@@ -94,21 +93,25 @@ export function listenOnce(element, eventType, listener, opt_capture) {
 
 /**
  * Returns  a promise that will resolve as soon as the specified event has
- * fired on the element. Optionally, opt_timeout can be specified that will
- * reject the promise if the event has not fired by then.
+ * fired on the element.
  * @param {!EventTarget} element
  * @param {string} eventType
  * @param {boolean=} opt_capture
- * @param {number=} opt_timeout
+ * @param {function(!UnlistenDef)=} opt_cancel An optional function that, when
```

Comment:
I thought about doing this, but it would be exposing it in the wrong object. Ie, how promise do not have a `#resolve` or `#reject`, only the resolver gets access to them. If we had a deferred type, we could easily add it there:

```js
/**
 * @typedef {{
 *   resolve: function(*=),
 *   reject: function(*=),
 *   cancel: function(),
 *   promise: !Promise,
 * }}
 */
let DeferredDef;
```